flang/lib/Semantics/canonicalize-omp.cpp

```diff
@@ -90,7 +90,11 @@ class CanonicalizationOfOmp {
     auto &dir{std::get<parser::OmpLoopDirective>(beginDir.t)};
 
     nextIt = it;
-    if (++nextIt != block.end()) {
+    while (++nextIt != block.end()) {
+      // Ignore compiler directives.
+      if (auto *directive{GetConstructIf<parser::CompilerDirective>(*nextIt)})
```

Comment:
I'm getting a warning on this line because the variable `directive` isn't used.  And since I have `-Werror` set, my build fails.

Reply:
Apologies for that. Neither my GCC nor my Clang builds that I run warned for this - Trivial fix here:

https://github.com/llvm/llvm-project/pull/81604